packages/astro/src/core/cookies/cookies.ts

```diff
@@ -0,0 +1,236 @@
+import type { CookieSerializeOptions } from 'cookie';
+import { parse, serialize } from 'cookie';
+import ms from 'ms';
+
+interface AstroCookieSetOptions {
+  domain?: string;
+  expires?: number | Date | string;
+  httpOnly?: boolean;
+  maxAge?: number;
+  path?: string;
+  sameSite?: boolean | 'lax' | 'none' | 'strict';
+  secure?: boolean;
+}
+
+interface AstroCookieDeleteOptions {
+	path?: string;
+}
+
+interface AstroCookieInterface {
+  value: string | undefined;
+  json(): Record<string, any>;
```

Comment:
If my previous comment is correct, this type would also be:

```suggestion
  json(): string | Record<string, any>;
```